topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorMetricsTest.java

```diff
@@ -159,67 +162,72 @@ public void batchingTopicControllerMetrics(KafkaCluster cluster) throws Interrup
             kt.getSpec().setConfig(Map.of("retention.ms", "86400000"));
             return kt;
         });
         controller.onUpdate(List.of(reconcilableTopic(t1ConfigChanged)));
         
-        // increase partitions (success)
+        // increase partitions, 1 reconciliation, success
         var t2PartIncreased = updateTopic("t2", kt -> {
             kt.getSpec().setPartitions(5);
             return kt;
         });
         controller.onUpdate(List.of(reconcilableTopic(t2PartIncreased)));
 
-        // decrease partitions (fail)
+        // decrease partitions, 1 reconciliation, fail
         var t2PartDecreased = updateTopic("t2", kt -> {
             kt.getSpec().setPartitions(4);
             return kt;
         });
         controller.onUpdate(List.of(reconcilableTopic(t2PartDecreased)));
 
-        // increase replicas (fail)
-        var t2ReplIncreased = updateTopic("t2", kt -> {
+        // increase replicas, 1 reconciliation, success
+        var t3ReplIncreased = updateTopic("t3", kt -> {
             kt.getSpec().setReplicas(2);
             return kt;
         });
-        controller.onUpdate(List.of(reconcilableTopic(t2ReplIncreased)));
+        controller.onUpdate(List.of(reconcilableTopic(t3ReplIncreased)));
 
-        // unmanage topic (success)
-        // should increase total and successful reconciliations metrics
+        // unmanage topic, 1 reconciliation, success
         var t1Unmanaged = updateTopic("t1", kt -> {
             kt.getMetadata().setAnnotations(Map.of(TopicOperatorUtil.MANAGED, "false"));
             return kt;
         });
         controller.onUpdate(List.of(reconcilableTopic(t1Unmanaged)));
         
-        // delete managed topics (success)
+        // delete managed topics, 1 reconciliation, success
         controller.onDelete(List.of(reconcilableTopic(t2)));
 
-        // delete unmanaged topic (success)
-        // should increase remove finalizer metric
+        // delete unmanaged topic, 1 reconciliation, success
         var t3Unmanaged = updateTopic("t3", kt -> {
             kt.getMetadata().setAnnotations(Map.of(TopicOperatorUtil.MANAGED, "false"));
             return kt;
         });
         controller.onDelete(List.of(reconcilableTopic(t3Unmanaged)));
 
         // check standard metrics
-        assertMetricMatches(TopicOperatorMetricsHolder.METRICS_RECONCILIATIONS, "counter", is(9.0));
-        assertMetricMatches(TopicOperatorMetricsHolder.METRICS_RECONCILIATIONS_SUCCESSFUL, "counter", is(7.0));
-        assertMetricMatches(TopicOperatorMetricsHolder.METRICS_RECONCILIATIONS_FAILED, "counter", is(2.0));
+        assertMetricMatches(TopicOperatorMetricsHolder.METRICS_RECONCILIATIONS, "counter", is(10.0));
+        assertMetricMatches(TopicOperatorMetricsHolder.METRICS_RECONCILIATIONS_SUCCESSFUL, "counter", is(9.0));
+        assertMetricMatches(TopicOperatorMetricsHolder.METRICS_RECONCILIATIONS_FAILED, "counter", is(1.0));
```

Comment:
It would be better to assert the metrics after each reconciliation.

Reply:
Definitely.